src/Mod/Sketcher/Gui/EditModeConstraintCoinManager.cpp

```diff
@@ -1976,12 +1975,13 @@ void EditModeConstraintCoinManager::rebuildConstraintNodes(
                 text->size.setValue(drawingParameters.labelFontSize);
                 text->lineWidth = 2 * drawingParameters.pixelScalingFactor;
                 text->useAntialiasing = false;
-                SoAnnotation* anno = new SoAnnotation();
-                anno->renderCaching = SoSeparator::OFF;
-                anno->addChild(text);
+                // Remove SoAnnotation wrapper to allow proper depth testing
+                // SoAnnotation* anno = new SoAnnotation();
+                // anno->renderCaching = SoSeparator::OFF;
+                // anno->addChild(text);

```

Comment:
Do we need to keep it? Also, that would mean that dimensions will be probably covered by gometry - not sure if that is intended or not.